driver-core/src/main/com/mongodb/ReadConcern.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2015 MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb;
+
+import static com.mongodb.assertions.Assertions.notNull;
+import static java.lang.String.format;
+
+/**
+ * The readConcern option allows clients to choose a level of isolation for their reads.
+ *
+ * The WiredTiger storage engine, introduces the readConcern option for replica sets and replica set shards.  Allowing clients to choose a
+ * level of isolation for their reads.
+ *
+ * @mongodb.server.release 3.2
+ * @since 3.2
+ */
+public enum ReadConcern {
```

Comment:
We should not make ReadConcern semantically equivalent to ReadConcernLevel.  Currently the level is all that is exposed, but that may change in the future.  Internally, the level is already one of two properties of the read concern (afterOpTime is the other, I believe).

See https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/ReadConcernLevel.cs and https://github.com/mongodb/mongo-csharp-driver/blob/master/src/MongoDB.Driver.Core/ReadConcern.cs for how I think this should be modelled.